feat(event-gateway): add image signature and attestation verification guide (#5645)
Kong/developer.konghq.com: app/_how-tos/event-gateway/verify-image-signatures-and-attestations.md

---
title: Verify {{site.event_gateway}} image signatures and attestations
content_type: how_to

permalink: /event-gateway/verify-image-signatures-and-attestations/

description: "Use Cosign to verify the signature of the {{site.event_gateway}} container image and to inspect the attestations attached to it, including the SBOM and vulnerability scans."

products:
    - event-gateway

works_on:
    - konnect

> 

Reviewer: ```suggestion

min_version:
  event-gateway: '1.1.1'
```

{:.info}
> The examples below use `kong/kong-event-gateway:{{ egw_release.version }}`. Replace the version with the {{site.event_gateway_short}} release you want to verify.
> The signing identity (`...@refs/tags/v{{ egw_release.version }}`) must match the exact release tag of the image you're verifying, so keep the version consistent between the image and the `--certificate-identity` flag.

## List the artifacts attached to an image

Use `cosign tree` to see every signature and attestation attached to an image:

```sh
cosign tree kong/kong-event-gateway:{{ egw_release.version }}
```

The output lists the supply chain artifacts (signatures and attestations) that are stored alongside the image in the registry.

## Verify the image signature

1. Read the manifest digest for the image with `regctl` and store it in a variable:

    ```sh
    export IMAGE_DIGEST=$(regctl manifest digest kong/kong-event-gateway:{{ egw_release.version }})
    echo $IMAGE_DIGEST
    ```

>     This captures the image's `SHA-256` digest so you can reuse it in the commands below, and prints it:

Reviewer: ```suggestion
    This captures the image's SHA-256 digest so you can reuse it in the commands below, and prints it:
```

    ```sh
    sha256:...
    ```
    {:.no-copy-code}

    Pinning to a digest guarantees you verify exactly the image you're going to run, even if the tag is later moved to a different image.

1. Verify the signature with `cosign verify`:

    ```sh
    cosign verify \
      --new-bundle-format=true \
      -a repo="kong-gateway/event-gateway" \
      -a workflow="CI" \
      --certificate-oidc-issuer="https://token.actions.githubusercontent.com" \
      --certificate-identity="https://github.com/kong-gateway/event-gateway/.github/workflows/ci.yaml@refs/tags/v{{ egw_release.version }}" \
      kong/kong-event-gateway@${IMAGE_DIGEST}
    ```

>     If verification is successful, the response contains a summary of the checks that were performed:

Reviewer: ```suggestion
    If verification succeeds, the response contains a summary of the checks that were performed:
```

      - Existence of the claims in the transparency log was verified offline
      - The code-signing certificate was verified using trusted certificate authority certificates
    ```
    {:.no-copy-code}

## Verify and inspect attestations

{{site.event_gateway}} images carry several signed attestations. The most commonly used predicate types are:

{% table %}
columns:
  - title: Attestation
    key: attestation
  - title: Predicate type
    key: predicate
  - title: "`--type` value"
    key: type
rows:
  - attestation: SPDX SBOM
    predicate: "`https://spdx.dev/Document`"
    type: "`spdxjson`"
  - attestation: CycloneDX SBOM
    predicate: "`https://cyclonedx.org/bom`"
    type: "`cyclonedx`"
  - attestation: Vulnerability scan (SARIF)
    predicate: "`https://cosign.sigstore.dev/sarif/vuln/...`"
    type: "`vuln`"
  - attestation: CIS Docker benchmark
    predicate: "`https://cisecurity.org/docker/...`"
    type: "_(use the full URL)_"
{% endtable %}

{:.info}
> The exact set of attestations can change between releases. Use [`cosign tree`](#list-the-artifacts-attached-to-an-image) to see the authoritative list for the image you're verifying.

> To verify an attestation's signature and print its contents, use `cosign verify-attestation` with the matching `--type`. For example, to verify the SPDX SBOM:

Reviewer: ```suggestion
To verify an attestation's signature and print its contents, use `cosign verify-attestation` with the matching `--type`. For example, to verify the SPDX SBOM, run:
```

```sh
cosign verify-attestation \
  --type="spdxjson" \
  --certificate-oidc-issuer="https://token.actions.githubusercontent.com" \
  --certificate-identity="https://github.com/kong-gateway/event-gateway/.github/workflows/ci.yaml@refs/tags/v{{ egw_release.version }}" \
  kong/kong-event-gateway@${IMAGE_DIGEST}
```

> For predicate types that don't have a built-in alias, pass the full predicate type URL to `--type` instead, for example `--type="https://cisecurity.org/docker/amd64"`.

Reviewer: ```suggestion
For predicate types that don't have a built-in alias, pass the full predicate type URL to `--type` instead. For example: `--type="https://cisecurity.org/docker/amd64"`.
```
